tensorflow/python/ops/array_ops.py

```diff
@@ -3628,7 +3645,114 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # Use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad(input=tensor, paddings=paddings, name=name)
+  start_coords, end_coords = [
+    shape(padded) - paddings[:,1] - shape(tensor), 
+    shape(padded) - paddings[:,1]
+    ]
+  tensor_to_copy = tensor
+  strides = ones(rank(padded))
+  # Add wraps for each dimension.
+  for i in range(shape(paddings)[0]):
```

Comment:
This is failing in graph mode:
```
Traceback (most recent call last):
  File "tensorflow/python/ops/array_ops_test.py", line 157, in testWrapPadOp
    array_ops._wrap_pad(t, paddings),
  File "tensorflow/python/ops/array_ops.py", line 3663, in _wrap_pad
    for i in range(shape(paddings)[0]):
TypeError: 'SymbolicTensor' object cannot be interpreted as an integer
```
This is essentially what I was warning below.  We don't know `shape(paddings)[0]` at graph-compile time when using dynamic shapes, so can't determine how to set up this loop.

You're probably going to need to use [`tf.while_loop`](https://www.tensorflow.org/api_docs/python/tf/while_loop) for loops, and [`tf.where`](https://www.tensorflow.org/api_docs/python/tf/where) for conditions.

Or you make assumptions/restrictions like the padding size must be smaller than the input so we can unconditionally overwrite the padding values.

Or, you do this all in C++ inside a kernel, where it _can_ do runtime conditionals/loops.  But then you might need to worry about support in XLA if you want to use this in a jit-compiled model.

Reply:
Ah thanks for taking the time to explain the various options! I'll probably have a look at `tf.while_loop` and `tf.where` and see if I can get this to work